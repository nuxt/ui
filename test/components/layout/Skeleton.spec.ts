import { describe, it, expect } from 'vitest'
import { USkeleton } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Skeleton', () => {
  it.each([
    [ 'basic case', { } ],
    [ '<USkeleton class="h-12 w-12" :ui="{ rounded: \'rounded-full\' }" />' ]
  ])('renders %s correctly', async (nameOrHtml: string, options?: TypeOf<typeof USkeleton.props>) => {
    const html = await ComponentRender(nameOrHtml, options, USkeleton)
    expect(html).toMatchSnapshot()
  })
})
