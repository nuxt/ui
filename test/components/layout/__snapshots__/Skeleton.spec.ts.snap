// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton > renders <USkeleton class="h-12 w-12" :ui="{ rounded: 'rounded-full' }" /> correctly 1`] = `"<div class="animate-pulse bg-gray-100 dark:bg-gray-800 rounded-full h-12 w-12"></div>"`;

exports[`Skeleton > renders basic case correctly 1`] = `"<div class="animate-pulse bg-gray-100 dark:bg-gray-800 rounded-md"></div>"`;
